src/libraries/System.Runtime.Extensions/tests/System/Random.cs

```diff
@@ -606,43 +606,23 @@ public void Xoshiro_AlgorithmBehavesAsExpected()
                     0xbe, 0x19, 0x0c, 0xf7, 0xae, 0xe2, 0xfa, 0xec, 0x8e, 0xc6, 0x22, 0xe1, 0x78, 0xb6, 0x39, 0xd1,
                 }, buffer);
 
-                Assert.Equal(50101881, randOuter.Next());
-                Assert.Equal(1272175254, randOuter.Next());
+                Assert.Equal(0.02333050652594726, randOuter.NextDouble());
+                Assert.Equal(0.592402766665435, randOuter.NextDouble());
+                Assert.Equal(0.6658883755588669, randOuter.NextDouble());
+                Assert.Equal(0.18356709019700435, randOuter.NextDouble());
+                Assert.Equal(0.8686094193061149, randOuter.NextDouble());
+
                 Assert.Equal(0, randOuter.Next(0));
                 Assert.Equal(0, randOuter.Next(1));
 
-                Assert.Equal(11, randOuter.Next(42));
```

Comment:
any reason you removed all these cases, rather than simply updating to whatever the new algorithm produces?

Reply:
Maybe I took the name of the test to literal: "Xoshiro_AlgorithmBehavesAsExpected"
From that I felt, that all it should do was to sufficiently verify, that the Xoshiro implementation was correct. From the referred link in the test https://www.pcg-random.org/posts/a-quick-look-at-xoshiro256.html it actually follows, that it is trivially predictable with only four outputs.

I could/can just put in the new values. But to me it is not a test of what the methods do, but of how they do it. It doesn't actually give any mark to whether or not the implementation is correct or not.

Reply:
incidentally, these tests already force a certain seed for reproducibility, I wonder whether someone else who implemented this algorithm (eg Swift until the last PR) has a set of test values that we could copy to make certain we're doing the same thing.

Reply:
> But to me it is not a test of what the methods do, but of how they do it. It doesn't actually give any mark to whether or not the implementation is correct or not.

It covers boundary cases, eg 
```
               Assert.Equal(2147483646, randOuter.Next(int.MaxValue - 5, int.MaxValue));
                Assert.Equal(-1841045958, randOuter.Next(int.MinValue, int.MaxValue));
```
That's never a bad idea, when it can be done so trivially.

Reply:
It also seems we lose meaningful coverage of public API eg  `NextInt64()` and I don't think this is enough for the overload
```
                Assert.Equal(0, randOuter.NextInt64(0));
                Assert.Equal(0, randOuter.NextInt64(1));
```

Reply:
If it makes you feel better, you could think of them as protection against some future subtle codegen regression on some random architecture :)

Reply:
Reverted to old style.
Seems like the testing style for swift is a bit different: https://github.com/apple/swift/blob/e1d1760c878990c675aa9a30ce5bf145c2d8abbb/validation-test/stdlib/Random.swift

Reply:
One or two of those might be worth porting at some point, eg., the Chi squared test, not suggesting you do it.

Historical context, the original/existing System.Random algorithm (still used for seeded constructor) as well as being very slow is also flawed due to an error made in the original code. This has been reported many times, with long discussion [here for example](https://github.com/dotnet/runtime/issues/23198). The decision was made to keep it for compatibility, as it's OK-ish, and fortunately the unseeded constructor is the most commonly used, so we were able to replace with a much faster algorithm producing a better distribution. I don't have easy access to whatever tests were written for System.Random (this was 20 years ago), but my guess is that they were simple functional tests, and not statistical tests, since the algorithm was just copied from Knuth (and assumed accurately so). I wonder whether some basic statistical tests could have discovered the bug way back then.

Reply:
I see Golang tests do statistical checks as well
https://github.com/golang/go/blob/master/src/math/rand/rand_test.go